server/src/models/Notes/NotesDocument.d.ts

```diff
@@ -0,0 +1,3 @@
+import mongoose from "mongoose";
+// eslint-disable-next-line
+export default interface NotesDocument extends mongoose.Document {}
```

Comment:
This should have all the fields of NotesCollection, see [here](https://thecodebarbarian.com/working-with-mongoose-in-typescript.html)

Also, an interface name, by convention, should begin with an I, so INotesDocument.